Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ElasticWindowMultiple.py

```diff
@@ -142,54 +144,68 @@ def PyExec(self):
             run_no = getInstrRun(input_ws)[1]
             run_numbers.append(run_no)
 
-            # Get the sample temperature
-            temp = self._get_temperature(input_ws)
-            if temp is not None:
-                temperatures.append(temp)
+            # Get the sample environment unit
+            sample,unit = self._get_sample_units(input_ws)
+            if sample is not None:
+                sample_param.append(sample)
             else:
-                # No need to output a tmperature workspace if there are no temperatures
+                # No need to output a temperature workspace if there are no temperatures
                 self._elt_workspace = ''
 
         logger.information('Creating Q and Q^2 workspaces')
+        progress.report('Creating Q workspaces')
 
         if len(input_workspace_names) == 1:
             # Just rename single workspaces
-            RenameWorkspace(InputWorkspace=q_workspaces[0],
-                            OutputWorkspace=self._q_workspace)
-            RenameWorkspace(InputWorkspace=q2_workspaces[0],
-                            OutputWorkspace=self._q2_workspace)
+            rename_alg = self.createChildAlgorithm("RenameWorkspace", enableLogging=False)
+            rename_alg.setProperty("InputWorkspace", q_workspaces[0])
+            rename_alg.setProperty("OutputWorkspace", self._q_workspace)
+            rename_alg.execute()
+            rename_alg.setProperty("InputWorkspace", q2_workspaces[0])
+            rename_alg.setProperty("OutputWorkspace", self._q2_workspace)
+            rename_alg.execute()
         else:
             # Append the spectra of the first two workspaces
-            AppendSpectra(InputWorkspace1=q_workspaces[0],
-                          InputWorkspace2=q_workspaces[1],
-                          OutputWorkspace=self._q_workspace)
-            AppendSpectra(InputWorkspace1=q2_workspaces[0],
-                          InputWorkspace2=q2_workspaces[1],
-                          OutputWorkspace=self._q2_workspace)
+            append_alg = self.createChildAlgorithm("AppendSpectra", enableLogging=False)
+            append_alg.setProperty("InputWorkspace1", q_workspaces[0])
+            append_alg.setProperty("InputWorkspace2", q_workspaces[1])
+            append_alg.setProperty("OutputWorkspace", self._q_workspace)
+            append_alg.execute()
+            mtd.addOrReplace(self._q_workspace, append_alg.getProperty("OutputWorkspace").value)
+            append_alg.setProperty("InputWorkspace1", q2_workspaces[0])
+            append_alg.setProperty("InputWorkspace2", q2_workspaces[1])
+            append_alg.setProperty("OutputWorkspace", self._q2_workspace)
+            append_alg.execute()
+            mtd.addOrReplace(self._q2_workspace, append_alg.getProperty("OutputWorkspace").value)
```

Comment:
Similar comment. We have the output properties and setting those will control if the workspace goes into the ADS so we don't and shouldn't add it manually.

I can see there are a few places where this type of line is done.